### ISSUE TITLE
Allowing dict key mismatch among dataloader, keras model output, loss, and metric

### ISSUE THREAD

**Issue by wuhy08** [tensorflow/tensorflow#45752]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.3.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**

Currently, when tf.keras.Model.call is called and return type is dictionary, it requires that the keys of the dictionary matches the keys of the Dataloader's sample output, which also need to be sames as the keys of losses and metrics during compile. 

When training a complex model, it might be the case that only certain branches of the model matches the dataloader, losses and metrics. It will be great if only the intersect of all the key sets are used, potentially with warning, instead of throwing exceptions like `The two structures don't have the same sequence length. Input structure has length 11, while shallow structure has length 10.`

**Will this change the current api? How?**

Potentially. During compilation phrase, a flag argument like `allow_key_mismatch` can be passed to suppress the error.

**Who will benefit with this feature?**

Users who train a complex model with a lot of branches (outputs used for different tasks)

**Any Other info.**

**Comment by ravikyram** [comment]
@wuhy08 

Do you have any use case that requires the feature you are interested in? Please feel free to submit a PR if you have use cases that supports that feature.Thanks!

**Comment by wuhy08** [comment]
Hi @ravikyram 
I will take a look at the code. Thank you!
